oh2.js

```diff
@@ -123,6 +130,77 @@ function turnOnOff(context, event) {
     rest.postItemCommand(event.payload.accessToken, event.payload.appliance.applianceId, state, success, failure);
 }
 
+function getLockState(context, event) {
+    var success = function (item) {
+        var header = {
+            messageId: event.header.messageId,
+            name: event.header.name.replace("Request", "Response"),
+            namespace: event.header.namespace,
+            payloadVersion: event.header.payloadVersion
+        };
+        var payload = {
+            lockState: item.state == "ON" ? "LOCKED" : "UNLOCKED"
+        };
+        var result = {
+            header: header,
+            payload: payload
+        };
+        utils.log('Done with result', JSON.stringify(result));
+        context.succeed(result);
+    };
+    var failure = function (error) {
+        context.done(null, utils.generateControlError(event.header.messageId, event.header.name, 'DependentServiceUnavailableError', error.message));
+    };
+    rest.getItem(event.payload.accessToken, event.payload.appliance.applianceId, success, failure);
+}
+
+function setLockState(context, event) {
+    var success = function (response) {
+        var header = {
+            messageId: event.header.messageId,
+            name: event.header.name.replace("Request", "Confirmation"),
+            namespace: event.header.namespace,
+            payloadVersion: event.header.payloadVersion
+        };
+        
+        var payload = {};
+        
+        var success = function (item) {
+            payload = {
+                lockState: item.state == "ON" ? "LOCKED" : "UNLOCKED"
+            };
+        };
+        
+        var failiure = function (error) {
+            payload = {
+                lockState: event.payload.lockState == "UNLOCKED" ? "LOCKED" : "UNLOCKED" //reverse to signify error
+            };
+        };
+
+        rest.getItem(event.payload.accessToken, event.payload.appliance.applianceId, success, failiure);
```

Comment:
> It may be permissible to add a timeout before verifying the state, but I don't know.

This could be $ costly for us, Lambda functions are charged by execution time, so we need to perform our actions quickly and return.   Still as I think about this, maybe having a 5 or ten second delay makes sense, or better have it poll the item once per second for 5 or ten seconds?  The usage should be fairly low on this feature, so maybe $'s are not a valid concern. 

BTW, I was not correct when I said that the call would return the state we just set, I had a small lapse in thinking :-)  Sending a command to a item will not necessarily set the state of that item, so the call may immediately return the last state ( without some type of wait/sleep/loop) .  Still the issue is the same.